task_processing/plugins/kubernetes/utils.py

```diff
@@ -144,3 +150,49 @@ def get_pod_volumes(volumes: PVector['DockerVolume']) -> List[V1Volume]:
         )
         for name, volume in unique_volumes.items()
     ]
+
+
+def get_node_affinity(affinities: PVector["NodeAffinity"]) -> Optional[V1NodeAffinity]:
+    # convert NodeAffinity into V1NodeSelectorRequirement
+    match_expressions = []
+    for aff in affinities:
+        op = aff["operator"]
+        val = aff["value"]
+
+        if op in {NodeAffinityOperator.IN, NodeAffinityOperator.NOT_IN}:
+            if type(val) != list:
+                raise TypeError(
+                    f"got non-list value '{val}' for affinity operator '{op}'"
+                )
+            val = [str(v) for v in val]
+
+        elif op in {NodeAffinityOperator.GT, NodeAffinityOperator.LT}:
+            if type(val) != int:
+                raise TypeError(
+                    f"got non-int value '{val}' for affinity operator '{op}'"
+                )
+            val = [str(val)]
+
+        elif op in {NodeAffinityOperator.EXISTS, NodeAffinityOperator.DOES_NOT_EXIST}:
+            val = []
+
+        else:
+            raise ValueError(
+                f"got invalid affinity operator '{op}'; "
+                f"expected one of: {NodeAffinityOperator.ALL}"
+            )
+
+        match_expressions.append(
+            V1NodeSelectorRequirement(key=str(aff["key"]), operator=op, values=val)
+        )
+
+    # package into V1NodeAffinity
+    if len(match_expressions) == 0:
```

Comment:
super-nit-feel-free-to-ignore:
```suggestion
    if not match_expressions:
```